src/monmove.cpp

```diff
@@ -114,36 +114,62 @@ bool monster::can_move_to( const tripoint &p ) const
         return false;
     }
 
-    // Various avoiding behaviors
-    if( has_flag( MF_AVOID_DANGER_1 ) || has_flag( MF_AVOID_DANGER_2 ) ) {
+    // Various avoiding behaviors.
+
+    bool avoid_fire = has_flag( MF_AVOID_FIRE );
+    bool avoid_fall = has_flag( MF_AVOID_FALL );
+    bool avoid_simple = has_flag( MF_AVOID_DANGER_1 );
+    bool avoid_complex = has_flag( MF_AVOID_DANGER_2 );
```

Comment:
All these bools could be const.

Reply:
One arguably could. The others can't, since they potentially change shortly thereafter, due to the cascading implications of the supersets.

Not without making it exceedingly ugly, anyway, and I think the benefit of the clarity of code far beats what it'd cost to make them const.